server/handle.go

```diff
@@ -151,14 +152,9 @@ func (l *listener4) HandleMsg4(buf []byte, oob *ipv4.ControlMessage, _peer net.A
 		} else if req.IsBroadcast() {
 			peer = &net.UDPAddr{IP: net.IPv4bcast, Port: dhcpv4.ClientPort}
 		} else {
-			// FIXME: we're supposed to unicast to a specific *L2* address, and an L3
-			// address that's not yet assigned.
-			// I don't know how to do that with this API...
-			//peer = &net.UDPAddr{IP: resp.YourIPAddr, Port: dhcpv4.ClientPort}
-			log.Warn("Cannot handle non-broadcast-capable unspecified peers in an RFC-compliant way. " +
-				"Response will be broadcast")
-
+			//sends a layer2 frame so that we can define the destination MAC address
 			peer = &net.UDPAddr{IP: net.IPv4bcast, Port: dhcpv4.ClientPort}
```

Comment:
https://tools.ietf.org/html/rfc2131#section-4.1(last paragraph) In this case this should be sent to the yiaddr, not broadcast
```suggestion
			peer = &net.UDPAddr{IP: resp.YourIPAddr, Port: dhcpv4.ClientPort}
```

Reply:
However that makes the logic under it to select the interface index incorrect, so I'd say lift the logic to choose an interface index out into its own function here